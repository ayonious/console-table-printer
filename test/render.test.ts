import { Table } from '../index';
import { getTableBody, getTableHeader } from './testUtils/getRawData';

describe('Table Rendering Tests', () => {
  it('should render basic table correctly', () => {
    const table = new Table()
      .addColumn('id')
      .addColumn('name')
      .addColumn('score');

    table.addRows([
      { id: 1, name: 'John', score: 85 },
      { id: 2, name: 'Jane', score: 92 },
      { id: 3, name: 'Bob', score: 78 }
    ]);

    const rendered = table.render();
    expect(rendered).toMatchSnapshot();
    table.printTable();
  });

  it('should render table with different alignments', () => {
    const table = new Table({
      columns: [
        { name: 'left_col', alignment: 'left' },
        { name: 'center_col', alignment: 'center' },
        { name: 'right_col', alignment: 'right' }
      ]
    });

    table.addRows([
      { left_col: 'Left', center_col: 'Center', right_col: 'Right' },
      { left_col: '111', center_col: '222', right_col: '333' }
    ]);

    const rendered = table.render();
    expect(rendered).toMatchSnapshot();
    table.printTable();
  });

  it('should render table with colors', () => {
    const table = new Table({
      columns: [
        { name: 'red_col', color: 'red' },
        { name: 'green_col', color: 'green' },
        { name: 'blue_col', color: 'blue' }
      ]
    });

    table.addRows([
      { red_col: 'Red text', green_col: 'Green text', blue_col: 'Blue text' },
      { red_col: '111', green_col: '222', blue_col: '333' }
    ]);

    const rendered = table.render();
    expect(rendered).toMatchSnapshot();
    table.printTable();
  });

  it('should render table with row colors', () => {
    const table = new Table()
      .addColumn('name')
      .addColumn('status')
      .addColumn('score');

    table.addRow(
      { name: 'John', status: 'Pass', score: 95 },
      { color: 'green' }
    );
    table.addRow(
      { name: 'Jane', status: 'Fail', score: 45 },
      { color: 'red' }
    );
    table.addRow(
      { name: 'Bob', status: 'Pass', score: 85 },
      { color: 'green' }
    );

    const rendered = table.render();
    expect(rendered).toMatchSnapshot();
    table.printTable();
  });

  it('should render table with mixed styling', () => {
    const table = new Table({
      columns: [
        { name: 'id', alignment: 'right', color: 'blue' },
        { name: 'name', alignment: 'left' },
        { name: 'status', alignment: 'center', color: 'green' }
      ]
    });

    table.addRow(
      { id: 1, name: 'John', status: 'Active' },
      { color: 'yellow' }
    );
    table.addRow(
      { id: 2, name: 'Jane', status: 'Inactive' },
      { color: 'gray' }
    );

    const rendered = table.render();
    expect(rendered).toMatchSnapshot();
    table.printTable();
  });

  it('should render empty table correctly', () => {
    const table = new Table({
      columns: [
        { name: 'col1' },
        { name: 'col2' }
      ]
    });

    const rendered = table.render();
    expect(rendered).toMatchSnapshot();
    table.printTable();
  });

  it('should render table with special characters', () => {
    const table = new Table()
      .addColumn('symbols')
      .addColumn('description');

    table.addRows([
      { symbols: '!@#$%^&*()', description: 'Special characters' },
      { symbols: '│─┌┐└┘├┤┬┴┼', description: 'Box drawing characters' },
      { symbols: '♠♣♥♦', description: 'Card suits' }
    ]);

    const rendered = table.render();
    expect(rendered).toMatchSnapshot();
    table.printTable();
  });

  it('should make sure each column is what its expected to be', () => {
    const table = new Table({
      shouldDisableColors: true,
      columns: [
        { name: 'id', alignment: 'right', color: 'blue' },
        { name: 'name', alignment: 'left' },
        { name: 'status', alignment: 'center', color: 'green' }
      ]
    });

    table.addRows([
      { id: 1, name: 'John', status: 'Active' },
      { id: 2, name: 'Jane', status: 'Inactive' }
    ]);

    const [renderedHeader, renderedBody] = [getTableHeader(table), getTableBody(table)];
    expect(renderedHeader).toEqual('│ id │ name │  status  │');
    expect(renderedBody).toEqual([
      '│  1 │ John │  Active  │',
      '│  2 │ Jane │ Inactive │'
    ]);
  });
});
