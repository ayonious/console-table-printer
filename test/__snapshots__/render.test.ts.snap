// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Rendering Tests should render basic table correctly 1`] = `
"┌────┬──────┬───────┐
│[37m [0m[01mid[0m │[37m [0m[01mname[0m │[37m [0m[01mscore[0m │
├────┼──────┼───────┤
│[37m [0m[37m 1[0m │[37m [0m[37mJohn[0m │[37m [0m[37m   85[0m │
│[37m [0m[37m 2[0m │[37m [0m[37mJane[0m │[37m [0m[37m   92[0m │
│[37m [0m[37m 3[0m │[37m [0m[37m Bob[0m │[37m [0m[37m   78[0m │
└────┴──────┴───────┘"
`;

exports[`Table Rendering Tests should render empty table correctly 1`] = `
"┌──────┬──────┐
│[37m [0m[01mcol1[0m │[37m [0m[01mcol2[0m │
├──────┼──────┤
└──────┴──────┘"
`;

exports[`Table Rendering Tests should render table with colors 1`] = `
"┌──────────┬────────────┬───────────┐
│[37m [0m[01m red_col[0m │[37m [0m[01m green_col[0m │[37m [0m[01m blue_col[0m │
├──────────┼────────────┼───────────┤
│[37m [0m[31mRed text[0m │[37m [0m[32mGreen text[0m │[37m [0m[34mBlue text[0m │
│[37m [0m[31m     111[0m │[37m [0m[32m       222[0m │[37m [0m[34m      333[0m │
└──────────┴────────────┴───────────┘"
`;

exports[`Table Rendering Tests should render table with different alignments 1`] = `
"┌──────────┬────────────┬───────────┐
│[37m [0m[01mleft_col[0m │[37m [0m[01mcenter_col[0m │[37m [0m[01mright_col[0m │
├──────────┼────────────┼───────────┤
│[37m [0m[37mLeft    [0m │[37m [0m[37m  Center  [0m │[37m [0m[37m    Right[0m │
│[37m [0m[37m111     [0m │[37m [0m[37m   222    [0m │[37m [0m[37m      333[0m │
└──────────┴────────────┴───────────┘"
`;

exports[`Table Rendering Tests should render table with mixed styling 1`] = `
"┌────┬──────┬──────────┐
│[37m [0m[01mid[0m │[37m [0m[01mname[0m │[37m [0m[01m status [0m │
├────┼──────┼──────────┤
│[37m [0m[34m 1[0m │[37m [0m[33mJohn[0m │[37m [0m[32m Active [0m │
│[37m [0m[34m 2[0m │[37m [0mJane │[37m [0m[32mInactive[0m │
└────┴──────┴──────────┘"
`;

exports[`Table Rendering Tests should render table with row colors 1`] = `
"┌──────┬────────┬───────┐
│[37m [0m[01mname[0m │[37m [0m[01mstatus[0m │[37m [0m[01mscore[0m │
├──────┼────────┼───────┤
│[37m [0m[32mJohn[0m │[37m [0m[32m  Pass[0m │[37m [0m[32m   95[0m │
│[37m [0m[31mJane[0m │[37m [0m[31m  Fail[0m │[37m [0m[31m   45[0m │
│[37m [0m[32m Bob[0m │[37m [0m[32m  Pass[0m │[37m [0m[32m   85[0m │
└──────┴────────┴───────┘"
`;

exports[`Table Rendering Tests should render table with special characters 1`] = `
"┌─────────────┬────────────────────────┐
│[37m [0m[01m    symbols[0m │[37m [0m[01m           description[0m │
├─────────────┼────────────────────────┤
│[37m [0m[37m !@#$%^&*()[0m │[37m [0m[37m    Special characters[0m │
│[37m [0m[37m│─┌┐└┘├┤┬┴┼[0m │[37m [0m[37mBox drawing characters[0m │
│[37m [0m[37m       ♠♣♥♦[0m │[37m [0m[37m            Card suits[0m │
└─────────────┴────────────────────────┘"
`;
